kafka-ui-react-app/src/components/Topics/Topic/Details/Details.tsx

```diff
@@ -19,16 +19,20 @@ import SettingsContainer from './Settings/SettingsContainer';
 interface Props extends Topic, TopicDetails {
   clusterName: ClusterName;
   topicName: TopicName;
+  state: RootState;
   deleteTopic: (clusterName: ClusterName, topicName: TopicName) => void;
   clearTopicMessages(clusterName: ClusterName, topicName: TopicName): void;
 }
 
 const Details: React.FC<Props> = ({
   clusterName,
   topicName,
+  state,
   deleteTopic,
   clearTopicMessages,
 }) => {
+  const isInternal = state.topics.byName[topicName].internal;
```

Comment:
We shouldn't pass whole state to the component. We should use selectors. In this case you can add 
```typescript
export const getIsTopicInternal = createSelector(
  getTopicByName,
  ({ internal }) => !!internal
);
```
selector in redux and use it in `DetailsContainer`. See `getTopicConfig` selector creation/usage example.

Reply:
done